converter/converters/koki_pod_to_kube_v1_pod.go

```diff
@@ -62,9 +62,204 @@ func Convert_Koki_Pod_to_Kube_v1_Pod(pod *types.PodWrapper) (*v1.Pod, error) {
+	if err != nil {
+		return nil, err
+	}
+	kubePod.Spec.ServiceAccountName = serviceAccount
+	kubePod.Spec.AutomountServiceAccountToken = autoMount
+	kubePod.Spec.NodeName = kokiPod.Node
+
+	net, pid, ipc, err := revertHostModes(kokiPod.HostMode)
+	if err != nil {
+		return nil, err
+	}
+	kubePod.Spec.HostNetwork = net
+	kubePod.Spec.HostPID = pid
+	kubePod.Spec.HostIPC = ipc
+	kubePod.Spec.ImagePullSecrets = revertRegistries(kokiPod.Registries)
+	kubePod.Spec.SchedulerName = kokiPod.SchedulerName
+
+	tolerations, err := revertTolerations(kokiPod.Tolerations)
+	if err != nil {
+		return nil, err
+	}
+	kubePod.Spec.Tolerations = tolerations
+
+	if kokiPod.FSGID != nil || kokiPod.GIDs != nil {
+		kubePod.Spec.SecurityContext = &v1.PodSecurityContext{}
+		kubePod.Spec.SecurityContext.FSGroup = kokiPod.FSGID
+		kubePod.Spec.SecurityContext.SupplementalGroups = kokiPod.GIDs
+	}
+
+	if kokiPod.Priority != nil {
+		kubePod.Spec.Priority = kokiPod.Priority.Value
+		kubePod.Spec.PriorityClassName = kokiPod.Priority.Class
+	}
+
 	return kubePod, nil
 }
 
+func revertTolerations(tolerations []types.Toleration) ([]v1.Toleration, error) {
+	var kubeTolerations []v1.Toleration
+
+	for i := range tolerations {
+		toleration := tolerations[i]
+		kubeToleration := v1.Toleration{
+			TolerationSeconds: toleration.ExpiryAfter,
+		}
+
+		fields := strings.Split(string(toleration.Selector), "=")
+		if len(fields) == 1 {
+			kubeToleration.Key = fields[0]
+			kubeToleration.Operator = v1.TolerationOpExists
+		} else if len(fields) == 2 {
+			kubeToleration.Key = fields[0]
+			kubeToleration.Operator = v1.TolerationOpEqual
+			kubeToleration.Value = fields[1]
+		} else {
+			return nil, util.TypeValueErrorf(toleration, "Unexpected toleration selector %s", toleration.Selector)
+		}
+
+		if kubeToleration.Value != "" {
+			fields := strings.Split(kubeToleration.Value, ":")
+			if len(fields) == 2 {
+				kubeToleration.Value = fields[0]
+				switch fields[1] {
+				case "NoSchedule":
+					kubeToleration.Effect = v1.TaintEffectNoSchedule
+				case "PreferNoSchedule":
+					kubeToleration.Effect = v1.TaintEffectPreferNoSchedule
+				case "NoExecute":
+					kubeToleration.Effect = v1.TaintEffectNoExecute
+				default:
+					return nil, util.TypeValueErrorf(toleration, "Unexpected toleration selector %s", toleration.Selector)
+				}
+			} else if len(fields) != 1 {
+				return nil, util.TypeValueErrorf(toleration, "Unexpected toleration effect %s", toleration.Selector)
+			}
+		}
+
+		kubeTolerations = append(kubeTolerations, kubeToleration)
+	}
+
+	return kubeTolerations, nil
+}
+
+func revertRegistries(registries []string) []v1.LocalObjectReference {
+	var kubeRegistries []v1.LocalObjectReference
+
+	for i := range registries {
+		ref := v1.LocalObjectReference{
+			Name: registries[i],
+		}
+		kubeRegistries = append(kubeRegistries, ref)
+	}
+
+	return kubeRegistries
+}
+
+func revertHostModes(modes []types.HostMode) (bool, bool, bool, error) {
```

Comment:
Use named return here since it's not obvious from the method signature which bool is which.